numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -55,6 +55,12 @@
  */
 #include "npy_math_private.h"
 
+static const npy_uint8  S[]   = {1, 2, 4, 8, 16};
+static const npy_uint8  B8[]  = {0x55,               0x33,               0x0F,               0x01};
+static const npy_uint16 B16[] = {0x5555,             0x3333,             0x0F0F,             0x0101};
+static const npy_uint32 B32[] = {0x55555555,         0x33333333,         0x0F0F0F0F,         0x01010101};
+static const npy_uint64 B64[] = {0x5555555555555555, 0x3333333333333333, 0x0F0F0F0F0F0F0F0F, 0x0101010101010101};
```

Comment:
What's `S` used for? Also, I think there's a case for adding a `npy_popcount(hh,h,,l,ll)` function to `npy_math.h`, and putting the tables there rather than here. In particular, we could then reuse the same function for implementing `bit_count` for `np.int32` scalars without having to go through ufunc overhead.

We use names of that form already here:

https://github.com/numpy/numpy/blob/47c5acf72571efe7cb731a51db594288e1dcc24c/numpy/core/src/npymath/npy_math_internal.h.src#L814

Reply:
Oh I see, let me try moving it.

Reply:
I have moved it in 975b8d1